src/formatDate.js

```diff
@@ -8,7 +8,70 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+  const dateParts = date.split(oldSeparator);
+  let year = '';
+  let month = '';
+  let day = '';
+  const formatDateParts = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i][0]) {
+      case 'Y':
+        year = dateParts[i];
+        break;
+      case 'M':
+        month = dateParts[i];
+        break;
+      case 'D':
+        day = dateParts[i];
+        break;
+      default:
+        throw new Error('not correct format');
+    }
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    switch (toFormat[i][0]) {
+      case 'Y':
+        formatDateParts.push(getFormatYear(year, toFormat[i]));
+        break;
+      case 'M':
+        formatDateParts.push(month);
+        break;
+      case 'D':
+        formatDateParts.push(day);
+        break;
+      default:
+        throw new Error('not correct format');
+    }
+  }
+
+  return formatDateParts.join(newSeparator);
+}
+
+/**
+ * @param {string} year
+ * @param {string} formatYear
+ *
+ * @returns {string}
+ */
+
+function getFormatYear(year, formatYear) {
+  if (formatYear.length === year.length) {
+    return year;
+  }
+
+  if (formatYear.length === 2) {
+    return year.slice(-2);
+  }
+
+  if (year < 30) {
+    return `20${year}`;
+  }
+
+  return `19${year}`;
```

Comment:
There's a logic error in this condition. The comparison `year < 30` will not work correctly because `year` is a string after being extracted from `dateParts`. You need to convert `year` to a number before this comparison. Additionally, if the year is between '00' and '29', you should prepend '20', but for '30' and above, prepend '19'. As it stands, the code will always prepend '19' to any two-digit year, which is incorrect for years '00' through '29'.